Delete more java only code

tasomaniac opened the pull request:
I realized we can delete even more :)

martinbonnin reviewed (approved):
It currently fails because `Optional` is still used from `apollo-runtime`. If everything is moved to Kotlin, we can use Kotlin optionals but there might be some Java files to convert.

tasomaniac commented:
Yeah, I thought this would be easier. I moved Optional to the runtime module. And converted some more unit tests to kotlin in order to remove this unnecessary abstraction over map.

martinbonnin commented on apollo-api/build.gradle.kts:
```diff
@@ -20,9 +20,7 @@ kotlin {
   }
 
-  jvm {
-    withJava()
```
Nice :)

sav007 reviewed (approved):
holly .... -1585 :)
